### ISSUE TITLE
No operator found for `memory_efficient_attention_backward`

### ISSUE THREAD

**Issue by KyleYueye** [facebookresearch/xformers#799]
# 🐛 Bug
xformers=0.0.16 is installed but a bug of No operator found for `memory_efficient_attention_backward` occurred

## To Reproduce

1. Training a controlnet, `xformers=0.0.16` is installed
2. `args.enable_xformers_memory_efficient_attention = True`
3. Error reported in `accelerator.backward(loss)`

```
if args.enable_xformers_memory_efficient_attention:
    if is_xformers_available():
        import xformers

        xformers_version = version.parse(xformers.__version__)
        if xformers_version == version.parse("0.0.16"):
            logger.warn(
                "xFormers 0.0.16 cannot be used for training in some GPUs. If you observe problems during training, please update xFormers to at least 0.0.17. See https://huggingface.co/docs/diffusers/main/en/optimization/xformers for more details."
            )
        unet.enable_xformers_memory_efficient_attention()
        controlnet.enable_xformers_memory_efficient_attention()
    else:
        raise ValueError("xformers is not available. Make sure it is installed correctly")
.......
.......
accelerator.backward(loss)
```
```
No operator found for `memory_efficient_attention_backward` with inputs: 
      query :       shape=(16, 1824, 1, 80) (torch.float32) 
      key :         shape=(16, 77, 1, 80) (torch.float32) 
      value :       shape=(16, 77, 1, 80) (torch.float32) 
      attn_bias :   <class 'NoneType'> 
      p :           0.0 
`flshattB` is not supported because: dtype=torch.float32 (supported: {torch.bfloat16, torch.float16}) 
          requires a GPU with compute capability == 8.0 for 'query.shape[-1] > 64' 

`cutlassB` is not supported because: Sm86 does not have enough shared-memory 
          to run this kernel - see https://github.com/facebookresearch/xformers/issues/517 

`smallkB` is not supported because: max(query.shape[-1] != value.shape[-1]) > 32 has custom scale 
          unsupported embed per head: 80
```
## Environment

Please copy and paste the output from the
environment collection script from PyTorch
(or fill out the checklist below manually).

You can run the script with:

```bash
# For security purposes, please check the contents of collect_env.py before running it.
python -m torch.utils.collect_env
```

- PyTorch Version (e.g., 1.0): `1.13.1+cu116`
- OS (e.g., Linux): `Ubuntu 20.04.4 LTS (x86_64)`
- How you installed PyTorch (`conda`, `pip`, source): `pip`
- Build command you used (if compiling from source): 
- Python version: `3.8`
- CUDA/cuDNN version: `11.6`
- GPU models and configuration: `GPU 0,1,2,3: NVIDIA GeForce RTX 3090`
- Any other relevant information:

## Additional context
```
xFormers 0.0.16
memory_efficient_attention.cutlassF:               available
memory_efficient_attention.cutlassB:               available
memory_efficient_attention.flshattF:               available
memory_efficient_attention.flshattB:               available
memory_efficient_attention.smallkF:                available
memory_efficient_attention.smallkB:                available
memory_efficient_attention.tritonflashattF:        available
memory_efficient_attention.tritonflashattB:        available
swiglu.fused.p.cpp:                                available
is_triton_available:                               True
is_functorch_available:                            False
pytorch.version:                                   1.13.1+cu116
pytorch.cuda:                                      available
gpu.compute_capability:                            8.6
gpu.name:                                          NVIDIA GeForce RTX 3090
build.info:                                        available
build.cuda_version:                                1107
build.python_version:                              3.8.16
build.torch_version:                               1.13.1+cu117
build.env.TORCH_CUDA_ARCH_LIST:                    5.0+PTX 6.0 6.1 7.0 7.5 8.0 8.6
build.env.XFORMERS_BUILD_TYPE:                     Release
build.env.XFORMERS_ENABLE_DEBUG_ASSERTIONS:        None
build.env.NVCC_FLAGS:                              None
build.env.XFORMERS_PACKAGE_FROM:                   wheel-v0.0.16
source.privacy:                                    open source
```

**Comment by danthe3rd** [comment]
Hi,
You should update xformers to a more recent version

**Comment by KyleYueye** [comment]
> Hi, You should update xformers to a more recent version

Sorry, but xformers >= 0.0.17 only support for torch 2.0
I have to use torch 2.0?
